In [1]:
import numpy as np
from scipy import stats

__Задача 1__

Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя уровень значимости 0.025 и правостороннюю критическую область, проверить данную гипотезу при условии, что в выборке из 100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

_Гипотеза $H_0$ утверждает, что $a = 17$_

_Дисперсия нам известна, $D(X) = 4$, $\sigma = 2$ соответственно. Будем использовать статистику $T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}}$_ 

_Нам нужно определить правостороннюю критическую область с $\alpha = 0.025$ для этой статистики и посмотреть, попадает ли наша статистика в критическую область_

In [3]:
mean = 17.5
a = 17
sig = 2
n = 100

T = (mean - a)/(sig / np.sqrt(n))
T

2.5

In [4]:
stats.norm.ppf(1 - 0.025)

1.959963984540054

_Значение нашей статистики попадает в критическую область, значит мы отвергаем нашу гипотезу. Утверждение $M(X) = 17$ неверно_

__Задача 2__

Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти доверительный интервал для оценки математического ожидания `a` с надёжностью 0.95, если выборочное среднее равно 80, а объём равен 256.

_Тут имеем $\sigma=16$, $p=0.95$, $\overline{X} = 80$ и $n=256$_

In [6]:
mean = 80
n = 256
std = 16
alpha = 0.05

t = stats.t.ppf(1 - alpha / 2, df=n - 1)
t

1.9693105698491928

_Имеем доверительный интервал:_

In [7]:
(mean - t * std / np.sqrt(n), mean + t * std / np.sqrt(n))

(78.0306894301508, 81.9693105698492)

_$P(78.03 \leq M(X) \leq 81.969) = 0.95%$_

__Задача 3__

Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:
```
202, 203, 199, 197, 195, 201, 200, 204, 194, 190
```
Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%?

In [29]:
samples = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
a = 200
alpha = 0.01

In [30]:
mean = samples.mean()
std = samples.std(ddof=1)

mean, std

(198.5, 4.453463071962462)

_Будем использовать статистику $T(X) = \dfrac{\overline{X} - a}{\sigma / \sqrt{n}}$_

In [31]:
T = (mean - a) / (std / np.sqrt(samples.shape[0]))
T

-1.0651074037450896

In [32]:
(stats.norm.ppf(alpha / 2), stats.norm.ppf(1 - alpha / 2))

(-2.575829303548901, 2.5758293035489004)

_Наша статистика показывает, что продавец не врет :)_

_Чтобы убедиться, построим доверительный интервал:_

In [33]:
t = stats.t.ppf(1 - alpha / 2, df=samples.shape[0] - 1)
t

3.2498355440153697

In [34]:
(mean - t * std / np.sqrt(samples.shape[0]), mean + t * std / np.sqrt(samples.shape[0]))

(193.92322840036354, 203.07677159963646)

_$a=200$ вполне укладывается в интервал_